kpatch-build/kpatch-elf.c

```diff
@@ -88,17 +88,29 @@ struct section *find_section_by_index(struct list_head *list, unsigned int index
 	return NULL;
 }
 
-struct section *find_section_by_name(struct list_head *list, const char *name)
+struct section *find_nth_section_by_name( struct list_head *list, int nth, const char *name)
 {
 	struct section *sec;
 
-	list_for_each_entry(sec, list, list)
-		if (!strcmp(sec->name, name))
-			return sec;
+	if (!list || !list->next || !name)
+		return NULL;
+
+	list_for_each_entry(sec, list, list) {
+		if (strcmp(sec->name, name))
+			continue;
+		if (--nth >= 0)
+			continue;
+		return sec;
+	}
```

Comment:
Why is this function needed?

Reply:
Ah, I think this is another piece of the arm64 port that turned out to not be needed.  (It was introduced in 17a90ecf9bb7629363e03100b5545bb254e03557 but then mostly reverted later in c48342cbf87fb8867114d98fef5b2487ed67bceb  Since this MR tries to squash that branch's journey of discovery, I think the original find_section_by_name should suffice w/o needing a nth search.)